servo/runner.py

```diff
@@ -368,6 +370,27 @@ def handle_progress_exception(error: Exception) -> None:
         )
         self.progress_handler_id = self.logger.add(self.progress_handler.sink, catch=True)
 
+        # Establish watch on the config file and shutdown when updated
+        # TODO: perform hot reload instead of shutdown for cases when running outside k8s and theres no pod to restart the exited container
+        async def _config_file_watch():
+            async for changes in watchgod.awatch(self.assembly.config_file, normal_sleep=1000):
+                # Only watching one file, should only ever get one change
+                if len(changes) > 1:
+                    self.logger.warning(f"servo.yaml config watch yielded multiple file changes: {changes}")
+                    change = next(filter(lambda c: c[1] == str(self.assembly.config_file), changes), None)
+                    if not change:
+                        continue
+                else:
+                    change = changes.pop()
+
+                self.logger.critical(f"Config file change detected ({str(change[0])}), shutting down active Servo(s) for config reload")
+                self.logger.trace(f"Config file watch change: {change}")
+
+                loop.create_task(self._shutdown(loop))
```

Comment:
I think we can do better here without too much effort.

We should be able to take the existing config and diff it against the incoming config and selectively restart the connectors that have had their configuration changed. We have a known mapping of configuration keys to in-memory objects.

I think we can even add/remove connectors and it shouldn't be that hard because we really only need to distill changes down to the top-level key affected.

I am very interested in this because I think where we want to go is clustering servos to provide HA, seamless upgrades, and keeping our resource utilization constrained at scale... I had 5000 servos running in one assembly back in October. Imagine if we just split that up across 5 instances with a gossip protocol paired with a Kubernetes mutating webhook admission controller to optimize everything as it is deployed